dev/bots/test.dart

```diff
@@ -333,6 +333,9 @@ Future<void> _flutterBuildIpa(String relativePathToApplication) async {
     await runCommand('pod',
       <String>['install'],
       workingDirectory: podfile.parent.path,
+      environment: <String, String>{
+        'LANG': '${Platform.localeName}.UTF8',
```

Comment:
Do we need a fallback if `Platform.localeName` is not set? I don't know if it'd happen for a CLI app, but when I was testing I noticed it wasn't set when I printed it inside a Flutter app, though it was set when running a CLI app (even when there was no `LANG` var set invoking the CLI app).

Reply:
https://github.com/dart-lang/sdk/blob/master/runtime/bin/platform_macos.cc#L190 Is the impl for macOS/iOS.  https://github.com/dart-lang/sdk/blob/master/runtime/bin/platform_android.cc#L121 android. 

These look like they shouldn't return null...  In fact,the main wrapper for it throws an error if they do.

Reply:
/cc @zanderso who probably knows more about that

Reply:
This is running on the iOS Simulator (iPhone), it gets `null`.

<img width="458" alt="Screenshot 2019-11-27 at 6 07 14 pm" src="https://user-images.githubusercontent.com/1078012/69748672-cb9f1100-1140-11ea-8055-56684529a7c6.png">

If it's only when running on-device, it might not matter (I can't imagine the flutter tool will be running pod on a device), but if there's another reason it's null (eg. that might occur for a real flutter_tool user on their host OS) we might need to guard against it (defaulting to `en_US` would presumably be fine).

Reply:
`Platform.localeName` can give `null` if the embedder overrides `_Platform._localeClosure` with a function that returns `null`. See: https://github.com/dart-lang/sdk/blob/master/sdk/lib/io/platform_impl.dart#L48.

The command line Dart VM doesn't override it, so `Platform.localeName` won't give `null`.

The engine does override it here: https://github.com/flutter/engine/blob/32ca0cc7d347235d8ae1ed53e49c4f0de97bd03c/lib/ui/dart_runtime_hooks.cc#L143

With: https://github.com/flutter/engine/blob/32ca0cc7d347235d8ae1ed53e49c4f0de97bd03c/lib/ui/hooks.dart#L68

Reply:
Got it - sounds like it'll always be set in for contexts where this code runs then.

Reply:
Actually, there are test failures are because this is null somewhere that's being called while testing:

```
00:06 +58: test/general.shard/persistent_tool_state_test.dart: state can be set and persists                                                                                                           NoSuchMethodError: The method 'replaceAll' was called on null.
Receiver: null
Tried calling: replaceAll("-", "_")
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      CocoaPods.cocoaPodsVersionText (package:flutter_tools/src/macos/cocoapods.dart:83:40)
```